interleave.py

```diff
@@ -0,0 +1,19 @@
+
+def interleave_generator(*args: iter):
+    """
+    A function that receives one or more parameters
```

Comment:
What are the types of the elements that it gets? (perhaps writing it here will make the documentation more informative).